rebasehelper/checkers/licensecheck_tool.py

```diff
@@ -0,0 +1,182 @@
+
+        diffs = []
+        for source_dir in [old_sources_path, new_sources_path]:
+            out = six.StringIO()
+            ProcessHelper.run_subprocess(["/usr/bin/licensecheck", source_dir, "--machine", "--recursive"],
+                                         output_file=out)
+            diff = {}
+            for l in out:
+                # licensecheck output format: 'Filepath\tlicense'
+                file_path, dlicense = l.split('\t')
+                # Get the important file path part. Strip ".../new_sources/package-{version}/..." part away
+                file_path = os.path.relpath(file_path, source_dir)
+                diff[file_path] = dlicense.strip()
+            diffs.append(diff)
+
+        old_lics, new_lics = set(), set()
+        changes = {'added': {}, 'changed': {}, 'removed': {}}
+        # Get changed licenses in existing files
+        for new_file, new_license in six.iteritems(diffs[1]):
+            new_lics.add(new_license)
+            for old_file, old_license in six.iteritems(diffs[0]):
+                old_lics.add(old_license)
+
+                if (new_file == old_file and
+                   (new_license != old_license)):
+                    new_key = '{} => {}'.format(old_license, new_license)
+                    if new_license == 'UNKNOWN':
+                        if old_license not in changes['removed']:
+                            changes['removed'][old_license] = []
+                        changes['removed'][old_license].append(new_file)
+                    elif old_license == 'UNKNOWN':
+                        if new_license not in changes['added']:
+                            changes['added'][new_license] = []
+                        changes['added'][new_license].append(new_file)
+                    else:
+                        if new_key not in changes['changed']:
+                            changes['changed'][new_key] = []
+                        if new_file not in changes['changed'][new_key]:
+                            changes['changed'][new_key].append(new_file)
+
+        # Get newly appeared files
+        for new_file, new_license in six.iteritems(diffs[1]):
+            if new_file not in diffs[0]:
+                if new_license not in changes['added']:
+                    changes['added'][new_license] = []
+                changes['added'][new_license].append(new_file)
+
+        # Get removed appeared files
+        for old_file, old_license in six.iteritems(diffs[0]):
+            if old_file not in diffs[1]:
+                if old_license not in changes['removed']:
+                    changes['removed'][old_license] = []
+                changes['removed'][old_license].append(old_file)
+
+        new_licenses = [lic for lic in new_lics if lic not in old_lics]
+        disappeared_licenses = [lic for lic in old_lics if lic not in new_lics]
+
+        if new_licenses or disappeared_licenses:
```

Comment:
Hm, how about taking advantage of sets and using:

```python
if old_lics ^ new_lics:
```
See: https://docs.python.org/3/library/stdtypes.html#frozenset.symmetric_difference